core/include/xfa.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (C) 2016 Kaspar Schleiser <kaspar@schleiser.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     core_util
+ * @brief       Cross File Arrays
+ * @{
+ *
+ * This macro, in combination with an entry in the linker scripts, allows the
+ * definition of constant arrays to be spread over multiple C compilation
+ * units. These arrays are called "cross-file arrays" or short xfa.
+ *
+ *
+ *
+ * @file
+ * @author      Kaspar Schleiser <kaspar@schleiser.de>
+ */
+
+#ifndef XFA_H
+#define XFA_H
+
+/*
+ * Unfortunately, current gcc trips over accessing XFA's because of their
+ * zero-size start/end array that are used of symbol markers, with an "array
+ * index out of bounds" warning.  So until a solution for that is found, we
+ * need to disable array bounds checks for files using XFAs.
+ */
+_Pragma("GCC diagnostic ignored \"-Warray-bounds\"")
+
+/**
+ * @brief helper macro for other XFA_* macros
+ *
+ * @internal
+ */
+#define _XFA(name, prio) __attribute__((used, section(".xfa." #name "." #prio)))
+
+/**
+ * @brief helper macro for other XFA_* macros
+ *
+ * @internal
+ */
+#define _XFA_CONST(name, prio) __attribute__((used, section(".roxfa." #name "." #prio)))
+
+/**
+ * @brief Define a read-only cross-file array
+ *
+ * This macro defines the symbols necessary to use XFA_START() and XFA_END().
+ * It needs to be part of one single compilation unit.
+ *
+ * @param[in]   type    name of the cross-file array
+ * @param[in]   name    name of the cross-file array
+ */
+#define XFA_INIT_CONST(type, name) \
+    _Pragma("GCC diagnostic push") \
+    _Pragma("GCC diagnostic ignored \"-Wpedantic\"") \
```

Comment:
What does the compiler complain about here?

Reply:
IIRC, zero sized arrays

Reply:
I double checked and added comments